src/python/aqueduct_executor/operators/utils/execution.py

```diff
@@ -0,0 +1,131 @@
+import io
+import sys
+import traceback
+
+from contextlib import redirect_stderr, redirect_stdout
+from typing import Callable, Optional
+from pydantic import BaseModel
+from aqueduct_executor.operators.utils.enums import ExecutionStatus, FailureType
+
+
+_GITHUB_ISSUE_LINK = "https://github.com/aqueducthq/aqueduct/issues/new?assignees=&labels=bug&template=bug_report.md&title=%5BBUG%5D"
+
+TIP_OP_EXECUTION = "Error executing operator. Please refer to the stack trace for fix."
+_TIP_CREATE_BUG_REPORT = (
+    f"Please create bug report in github: {_GITHUB_ISSUE_LINK} . "
+    "We will get back to you as soon as we can."
+)
+TIP_UNKNOWN_ERROR = f"Sorry, we've run into an unexpected error! {_TIP_CREATE_BUG_REPORT}"
+TIP_INTEGRATION_CONNECTION = (
+    "We have trouble connecting to the integration. "
+    "Please check your credentials or contact your integraiton provider."
+)
+TIP_DEMO_CONNECTION = f"We have trouble connecting to demo DB. {_TIP_CREATE_BUG_REPORT}"
+
+TIP_EXTRACT = "We couldn't execute the provided query. Please double check your query is correct."
+TIP_LOAD = "We couldn't load to the integration. Please make sure the target exists, or you have the right permission."
+TIP_DISCOVER = "We couldn't list items in the integration. Please make sure your credentials have the right permission."
+
+
+class Error(BaseModel):
+    context: str = ""
+    tip: str = ""
+
+
+class Logs(BaseModel):
+    stdout: str = ""
+    stderr: str = ""
+
+
+class ExecutionState(BaseModel):
+    user_logs: Logs
+    status: ExecutionStatus = ExecutionStatus.PENDING
+    failure_type: FailureType = FailureType.SUCCESS
```

Comment:
it seems strange to have a `FailureType.SUCCESS`. Any reason this can't just be an `Optional[FailureType]`, and anyone using ExecutionState can just check `ExecutionState.failure_type is not None`?

Reply:
or I guess they would just check if `ExecutionState.error is not None`

Reply:
Can you note in a comment here what each of these fields means actually? Whats the source of truth of whether an execution state succeeded? Does an error != None also mean that `failure_type` is also none (or SUCCESS), and that status must be `FAILED`? The relationship between all these fields may be unclear to someone not familiar with all  the code.

Reply:
I think `Optional[FailureType]` is a good idea which avoids a lot of confusion